app/frontend/src/features/profile/hooks/useUpdateHostingPreferences.test.tsx

```diff
@@ -41,25 +41,22 @@ describe("useUpdateHostingPreference hook", () => {
   it("updates the store with the latest user hosting preference", async () => {
     addDefaultUser();
 
-    const newUserPref = (Object.keys(
-      newHostingPreferenceData
-    ) as (keyof HostingPreferenceData)[]).reduce(
-      (acc, key: keyof HostingPreferenceData) => {
-        switch (key) {
-          case "smokingAllowed":
-          case "parkingDetails":
-          case "sleepingArrangement":
-            acc[key] = newHostingPreferenceData[key];
-            return acc;
-          default:
-            acc[key] = {
-              value: newHostingPreferenceData[key],
-            };
-            return acc;
-        }
-      },
-      {} as Record<string, unknown>
-    );
+    const newUserPref = (Object.entries(newHostingPreferenceData) as [
+      keyof HostingPreferenceData,
+      string | number | boolean | null
+    ][]).reduce((acc, [key, value]) => {
+      switch (key) {
+        case "smokingAllowed":
+        case "parkingDetails":
+        case "sleepingArrangement":
+          acc[key] = value;
+          return acc;
+        default:
+          acc[key] = { value: value };
+          return acc;
+      }
+    }, {} as Record<string, string | number | boolean | null | { value: string | number | boolean | null }>);
```

Comment:
```suggestion
    type Primitives = string | number | boolean | null;
    type HostPreferenceValue = Primitives | { value: Primitives };
    const newUserPref = Object.entries(newHostingPreferenceData).reduce(
      (acc: Record<string, HostPreferenceValue>, [key, value]) => {
        switch (key) {
          case "smokingAllowed":
          case "parkingDetails":
          case "sleepingArrangement":
            acc[key] = value;
            return acc;
          default:
            acc[key] = { value };
            return acc;
        }
      },
      {}
    );
```
I think annotating the accumulator object and type aliasing it makes the types a bit neater to read and eliminates most needs for type casting.

Reply:
Yes, that looks better! Except, I belive the type for HostPreferenceValue should be: 
`Primitives | { value: number }`

Since I think this type can be used on other places as well, should I add the following to `utils/types.ts`?
```
export type Primitives = string | number | boolean | null;
export type ServicePrimitives = Primitives | { value: number };
```

Or what is a better name for the `ServicePrimitives` type?

Reply:
But your suggestion still drops the type safety (and intellisense) for the properties of type `HostingPreferenceData`.

```
const newUserPref = (Object.entries(newHostingPreferenceData) as [
      keyof HostingPreferenceData,
      Primitives
    ][]).reduce(
      (acc, [key, value]) => {
        switch (key) {
          case "smokingAllowed":
          case "parkingDetails":
          case "sleepingArrangement":
            acc[key] = value;
            return acc;
          default:
            if (typeof value !== "number") {
              throw new Error(`Value of enum ${key} must be a number.`)
            }
            acc[key] = { value };
            return acc;
        }
      },
      {} as Record<string, ServicePrimitives>
    );
```

Using the `keyof HostingPreferenceData` type gives the following intellisense in vscode:

![image](https://user-images.githubusercontent.com/4201102/115195603-3dd61400-a0ef-11eb-8633-33c8e5adf0ae.png)

and the following added type safety which I think is needed to prevent errors after properties either renamed or removed:

![image](https://user-images.githubusercontent.com/4201102/115195848-8ab9ea80-a0ef-11eb-875e-871602d7a6ba.png)